core/src/main/scala/kafka/server/DelayedOperation.scala

```diff
@@ -46,6 +46,11 @@ import scala.collection.mutable.ListBuffer
 abstract class DelayedOperation(override val delayMs: Long) extends TimerTask with Logging {
 
   private val completed = new AtomicBoolean(false)
+  protected var lock: ReentrantLock = null
+
+  def createLock(): Unit = {
+    lock = new ReentrantLock
```

Comment:
Is the lazy creation worth it? I'd imagine that the other state in the operation would be much larger than the lock (we typically have a map of partitions to some value). If we decide that keeping it lazy is worth it, don't we have to make it `volatile`?

Reply:
Also, if we keep this approach, we should tighten the access modifiers. `createLock` doesn't need to be public, we should make the `var` private and expose a protected `lock` accessor, etc.

Reply:
Remove lazy initialization.